### ISSUE TITLE
@Format and @QueryValue should fail when an invalid LocalDate is sent

### ISSUE THREAD

**Issue by sbeins55** [micronaut-projects/micronaut-core#6293]
### Expected Behavior

When sending an invalid date string to a controller with `@Format` the controller should respond with a Bad Request if the QueryValue is not valid for the expected type.

### Actual Behaviour

Invalid dates can be downgraded `2021-02-31 -> 2021-02-28` or just become null `2021-02-32 -> null`

### Steps To Reproduce

The repo below shows this behavior in the test cases.

1. Create a controller endpoint with a query parameter using `@Format("yyyy-MM-dd") @QueryValue test: LocalDate`
2. Send an invalid LocalDate string
3. See some dates are downgraded or forced to be null

### Environment Information

- Operating System: macOs BigSur version 11.5.2
- JDK Version: 11

### Example Application

https://github.com/sbeins55/date-parsing

### Version

3.0.3

**Comment by sbeins55** [comment]
I wouldn't mind contributing a PR for this as I think I found what's happening. 

It looks like the [type converter uses a SimpleDateFormat](https://github.com/micronaut-projects/micronaut-core/blob/3.0.x/core/src/main/java/io/micronaut/core/convert/DefaultConversionService.java#L953) which by default uses lenient parsing. The lenient parsing seems to be causing the auto downgrades. 

Only reason I'm hesitant to open a PR is I'm no longer 100% certain that an invalid query parameter should throw a bad request, but maybe make it null? I'm also thinking lenient parsing is sometimes desired so this should be a configuration with the default being the current behavior?

**Comment by graemerocher** [comment]
bad request makes sense to me unless it is `@Nullable` 

PRs welcome

**Comment by jameskleeh** [comment]
@sbeins55 Agreed - making it configurable with the default being the current behavior

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?

**Comment by graemerocher** [comment]
Triage: the original reproducer identified lenient `SimpleDateFormat` parsing, but current `context` conversion uses `DateTimeFormatter` for `LocalDate` and explicit `@Format` parsing, which rejects invalid calendar dates instead of silently normalizing them. This addresses the reported `2021-02-31`/`2021-02-32` behavior. Closing as completed; a current regression should include the target type and conversion path if it still accepts invalid dates.
